internal/component/database_observability/postgres/collector/pg_activity.go

```diff
@@ -0,0 +1,373 @@
+	level.Debug(c.logger).Log("msg", ActivityName+" collector started")
+
+	c.running.Store(true)
+	ctx, cancel := context.WithCancel(ctx)
+	c.ctx = ctx
+	c.cancel = cancel
+
+	go func() {
+		defer func() {
+			c.Stop()
+			c.running.Store(false)
+		}()
+
+		ticker := time.NewTicker(c.collectInterval)
+
+		for {
+			if err := c.fetchActivity(c.ctx); err != nil {
+				level.Error(c.logger).Log("msg", "collector error", "err", err)
+			}
+
+			select {
+			case <-c.ctx.Done():
+				return
+			case <-ticker.C:
+				// continue loop
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (c *Activity) Stopped() bool {
+	return !c.running.Load()
+}
+
+// Stop should be kept idempotent
+func (c *Activity) Stop() {
+	c.cancel()
+}
+
+// calculateDuration returns a formatted duration string between a nullable time and current time
+func calculateDuration(nullableTime sql.NullTime, currentTime time.Time) string {
+	if nullableTime.Valid {
+		return currentTime.Sub(nullableTime.Time).Round(time.Millisecond).String()
+	}
+	return ""
+}
+
+func (c *Activity) fetchActivity(ctx context.Context) error {
+	slog.Info("Fetching activity")
+	scrapeTime := time.Now()
+	rows, err := c.dbConnection.QueryContext(ctx, selectPgStatActivity, c.lastScrape)
+	if err != nil {
+		level.Error(c.logger).Log("msg", "failed to query pg_stat_activity", "err", err)
+		return err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		activity := ActivityInfo{}
+		err := rows.Scan(
+			&activity.DatabaseName,
+			&activity.DatabaseID,
+			&activity.PID,
+			&activity.LeaderPID,
+			&activity.UserSysID,
+			&activity.Username,
+			&activity.ApplicationName,
+			&activity.ClientAddr,
+			&activity.ClientPort,
+			&activity.StateChange,
+			&activity.Now,
+			&activity.BackendStart,
+			&activity.XactStart,
+			&activity.QueryStart,
+			&activity.WaitEventType,
+			&activity.WaitEvent,
+			&activity.State,
+			&activity.BackendType,
+			&activity.BackendXID,
+			&activity.BackendXmin,
+			&activity.QueryID,
+			&activity.Query,
+			&activity.BlockedByPids,
+		)
+		if err != nil {
+			level.Error(c.logger).Log("msg", "failed to scan pg_stat_activity set", "err", err)
+			continue
+		}
+
+		_, err = c.validateActivity(activity)
+		if err != nil {
+			level.Debug(c.logger).Log("msg", "invalid pg_stat_activity set", "err", err)
+			continue
+		}
+
+		query := ""
+		if activity.Query.Valid {
+			query = activity.Query.String
+		}
+
+		leaderPID := ""
+		if activity.LeaderPID.Valid {
+			leaderPID = fmt.Sprintf(`%d`, activity.LeaderPID.Int64)
+		}
+
+		stateDuration := calculateDuration(activity.StateChange, activity.Now)
+		queryDuration := calculateDuration(activity.QueryStart, activity.Now)
+		xactDuration := calculateDuration(activity.XactStart, activity.Now)
+		backendDuration := calculateDuration(activity.BackendStart, activity.Now)
+
+		databaseName := ""
+		if activity.DatabaseName.Valid {
+			databaseName = activity.DatabaseName.String
+		}
+
+		userName := ""
+		if activity.Username.Valid {
```

Comment:
`activity.Username.String` will return either the value, or an empty string if not valid. I don't think we're gaining anything from putting these `.Valid` checks in